### ISSUE TITLE
Capabilities setting/editing for post types and taxonomies.

### ISSUE THREAD

**Issue by tw2113** [WebDevStudios/custom-post-type-ui#390]

**Comment by lustremedia** [comment]
Is there a workaround to this? I saw the filtering for the taxonomies here https://wordpress.org/support/topic/want-to-manage-the-custom-taxonomy-capabilities/ and was wondering if something similar existed for post_type?

Basically I just want to fade out the "add new" button for a couple of post types, as explained here:
https://stackoverflow.com/questions/3235257/wordpress-disable-add-new-on-custom-post-type/16675677#16675677

**Comment by tw2113** [comment]
Yes, for majority of the hooks that we add to our plugins, especially when related to details for post types or taxonomies, there's a similar hook for both.

For hooking in and amending arguments for a post type, please refer to https://github.com/WebDevStudios/custom-post-type-ui/blob/1.5.7/custom-post-type-ui.php#L441-L453

**Comment by lustremedia** [comment]
@tw2113 Hi yes, I saw that, however I am not quite sure about all the input parameters.  Could you maybe provide an example for post types as needed in functions.php as you have done here for taxonomies? -> https://wordpress.org/support/topic/want-to-manage-the-custom-taxonomy-capabilities/

**Comment by tw2113** [comment]
It's going to be pretty much the exact same structure.

```
function lustremedia_customize_post_type( $args, $post_type_slug ) {

	if ( 'some_slug' === $post_type_slug ) {
		$args['capability_type'] => $post_type_slug,
		$args['capabilities'] = array(
			'create_posts' => false,
		);
		// map_meta_cap is already set to true by default for CPTUI
	}

	return $args;
}
add_filter( 'cptui_pre_register_post_type', 'lustremedia_customize_post_type', 10, 2 );
```

Adjust as needed from here.

**Comment by lustremedia** [comment]
I just figured it out myself ... 

```
function remove_add_new( $args, $post_type_slug ) {
	if ( 'my_slug' === $post_type_slug ) {
		$args['capabilities'] = array(
			'create_posts' => 'do_not_allow',
		);
	}
	return $args;
}
add_filter( 'cptui_pre_register_post_type', 'remove_add_new', 10, 2 );
```
needs to be this, I guess false is deprecated ... : 'create_posts' => 'do_not_allow', 

Cheers!

**Comment by KZeni** [comment]
Picking up what was discussed at https://github.com/tw2113/custom-post-type-ui-capabilities/issues/3, I think there are still a good number of benefits to integrating this separate plugin (https://github.com/tw2113/custom-post-type-ui-capabilities) into the main plugin.

Currently, this offering is very difficult to find when it really could be quite nice/useful for those that need it while it's also not really being kept up-to-date with the rest of the plugin since that then needs to be done separately (when those updates, like minor UI adjustments, would otherwise be really quick.)

Meanwhile, the concern about visual noise & most users not needing this feature (while a valid concern) could be alleviated by having something like "Enable Capability Mapping" as a checkbox which then toggles these additional settings on the page. Users that don't need it just see it as one more checkbox in the list of settings that they can ignore while those that do need it have it right at their fingertips rather than assuming something like this isn't available unless they happen to stumble across this on the Help/Support page of CPT UI (and are then also okay with a non WP.org directory plugin [making updates more of a concern] that then might appear to be abandoned or otherwise have questionable support [per this part of things maybe not having any updates in years.])

I'm just thinking this decision could potentially be reconsidered & thanks for hearing me out. 😊
